decode_amd64.s

```diff
@@ -184,9 +184,7 @@ tagLit60Plus:
 	// checks. In the asm version, we code it once instead of once per switch case.
 	ADDQ CX, SI
 	SUBQ $58, SI
-	MOVQ SI, BX
-	SUBQ R11, BX
-	CMPQ BX, R12
+	CMPQ SI, R13
```

Comment:
Is it possible that (SI + CX - $58) might have overflowed, if R11 (src_base) was a very high pointer? I think that the subsequent subtraction of R11 used to eliminate that possibility. If overflow is possible with your change, is there a cheap fix?

Reply:
Good point.

Valid pointers should be no more than 48 bits on AMD64. RCX is between $60-$63.

So unless the function gets called with either invalid pointer or length it shouldn't be possible to overflow RSI. 

But if you know someone with more detailed experience in this, a second review could of course be fine.

Reply:
Wikipedia isn't official documentation, but https://en.wikipedia.org/wiki/X86-64#Virtual_address_space_details does say "the AMD specification requires that the most significant 16 bits of any virtual address, bits 48 through 63, must be copies of bit 47 (in a manner akin to sign extension)". If I'm reading that correctly, 0xFFFF_FFFF_FFFF_FFF0 could be a valid pointer value, even though only the low 48 bits are 'used'.